openquakeplatform/test/oqplatform.py

```diff
@@ -0,0 +1,425 @@
+        #<i class="icon-off"></i>
+        #Log out
+        #</a>
+        logout_button = self.xpath_finduniq(
+            "//a[@href='/account/logout/' and normalize-space(text())"
+            " = 'Log out']")
+
+        logout_button.click()
+
+        #check new url
+        self.wait_new_page(logout_button, '/account/logout/')
+
+        #<button class="btn btn-primary" type="submit">Log out</button>
+        logout_button = self.xpath_finduniq(
+            "//button[@type='submit' and normalize-space("
+            "text()) = 'Log out']")
+        logout_button.click()
+
+        self.driver.quit()
+
+    def cleanup(self):
+        # removes secondary platforms
+        platforms_loop = self.platforms[:]
+        for platform in platforms_loop:
+            self.platform_destroy(platform)
+
+        users_loop = self.users[:]
+        for user in users_loop:
+            self.user_del(user)
+
+    def navigate(self, dest):
+        # driver, basepath, self.dest):
+        if dest not in ["calc", "explore", "share"]:
+            raise ValueError
+
+        try:
+            dest_button = self.xpath_finduniq("//a[@href='/%s/']" % dest)
+        except (TimeoutError, ValueError, NotUniqError):
+            self.driver.get(self.basepath)
+            dest_button = self.xpath_finduniq("//a[@href='/%s/']" % dest)
+
+        dest_button.click()
+        self.wait_new_page(dest_button, '/%s/' % dest)
+
+    def get(self, url):
+        self.driver.get(self.basepath + url)
+
+    def xpath_finduniq(self, xpath_str, times=50, postfind=0.1):
+        for t in range(0, times):
+            field = self.driver.find_elements(By.XPATH, xpath_str)
+            if len(field) > 0:
+                break
+            if times > 1:
+                time.sleep(self.DT)
+        else:
+            if times > 1:
+                raise TimeoutError(
+                    "Timeout waiting '{}' for {} seconds.".format(
+                        xpath_str, times * self.DT)
+                    )
+            else:
+                raise ValueError(
+                    "Search path '{}' not matches.".format(xpath_str)
+                    )
+
+        if len(field) > 1:
+            raise NotUniqError(
+                "Waiting for '{}' returned {} matches.".format(xpath_str,
+                                                               len(field))
+                )
+
+        if postfind > 0:
+            time.sleep(postfind)
+        return field[0]
+
+    def wait_new_page(self, element, url):
+        from selenium.common.exceptions import StaleElementReferenceException
+
+        def link_has_gone_stale():
+            try:
+                # poll the link with an arbitrary call
+                element.find_elements_by_id('doesnt-matter')
+                return False
+            except StaleElementReferenceException:
+                if (self.driver.current_url == url
+                        or self.driver.current_url == (self.basepath + url)):
```

Comment:
Pep8 tool is not following latest pep8 rules, we already do it:
  https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator